src/test/shared/sam/cli/testSamCliProcessInvoker.ts

```diff
@@ -73,51 +71,98 @@ export class FakeChildProcessResult implements ChildProcessResult {
-        {
-            exitCode = 0,
-            stdout = '',
-            stderr = '',
-            ...params
-        }: Partial<ChildProcessResult>
-    ) {
+    public constructor({ exitCode = 0, stdout = '', stderr = '', ...params }: Partial<ChildProcessResult>) {
         this.exitCode = exitCode
         this.error = params.error
         this.stdout = stdout
         this.stderr = stderr
     }
 }
 
-export function assertErrorContainsBadExitMessage(
-    actualError: Error,
-    sourceErrorMessage: string
-) {
+export function assertErrorContainsBadExitMessage(actualError: Error, sourceErrorMessage: string) {
     assert.strictEqual(
-        actualError.message, `Error with child process: ${sourceErrorMessage}`,
+        actualError.message,
+        `Error with child process: ${sourceErrorMessage}`,
         'Unexpected error message'
     )
 }
 
 export async function assertLogContainsBadExitInformation(
     logger: TestLogger,
     errantChildProcessResult: ChildProcessResult,
-    expectedExitCode: number,
+    expectedExitCode: number
 ): Promise<void> {
-    assert.ok(
-        // tslint:disable-next-line:max-line-length
-        await logger.logContainsText(`Unexpected exitcode (${errantChildProcessResult.exitCode}), expecting (${expectedExitCode})`),
-        'Log message missing for exit code'
-    )
-    assert.ok(
-        await logger.logContainsText(`Error: ${errantChildProcessResult.error}`),
-        'Log message missing for error'
-    )
-    assert.ok(
-        await logger.logContainsText(`stderr: ${errantChildProcessResult.stderr}`),
-        'Log message missing for stderr'
-    )
-    assert.ok(
-        await logger.logContainsText(`stdout: ${errantChildProcessResult.stdout}`),
-        'Log message missing for stdout'
-    )
+    const logPath = logger.logPath
+
+    // Give the log a chance to get created/flushed
+    await retryOnError({
+        description: 'Wait for log file to exist',
+        fn: async () => await access(logPath),
+        retries: 20,
+        delayMilliseconds: 50,
+        failureMessage: `Could not find log file: ${logPath}`
+    })
+    await retryOnError({
+        description: 'Wait for expected text to appear in log',
+        fn: async () => {
+            const logText = await readFileAsString(logPath)
+            const verifyMessage = verifyLogText(logText, errantChildProcessResult, expectedExitCode)
+            if (verifyMessage) {
+                console.log(verifyMessage)
+                throw new Error(verifyMessage)
+            }
+        },
+        retries: 20,
+        delayMilliseconds: 50,
+        failureMessage: 'Did not find expected log contents'
+    })
+}
+
+function verifyLogText(
+    text: string,
+    errantChildProcessResult: ChildProcessResult,
+    expectedExitCode: number
+): string | undefined {
+    const expectedText = [
+        {
+            text: `Unexpected exitcode (${errantChildProcessResult.exitCode}), expecting (${expectedExitCode})`,
```

Comment:
imo this is kind of useless as a utility function since it has the text hard coded into it, maybe that should move out and be another argument

Reply:
Thanks, I needed that fresh set of eyes there